src/cf.data.pre

```diff
@@ -1275,9 +1275,14 @@ ENDIF
 	  # Referer is highly unreliable, so use with care
 
 	acl aclname ident [-i] username ...
+	  # string match on the ident user identifier [slow]
+	  # use REQUIRED to accept any non-null user identifier.
 	acl aclname ident_regex [-i] pattern ...
-	  # string match on ident output [slow]
-	  # use REQUIRED to accept any non-null ident.
+	  # regex match against the ident user identifier [slow]
+	  # use a dash ('-') to match an invalid ident reply, including cases
+	  # when the user identifier is missing.
+	  # NOTE: a side effect: a dash ('-') will also match the user identifier
+	  # consisting of a single dash.
```

Comment:
Please clarify whether dash applies to cases where the user identifier was not even requested. I also think that we should document that this ACL triggers transactions with an Authentication Server. For example (shown here with poor formatting):

```suggestion
	  # regex match against the ident user identifier [slow]
	  #
	  # If necessary, evaluation of this ACL triggers a transaction with an Authentication Server
	  # (a.k.a. Ident server; see RFC 1413) to If identity the user behind client-to-Squid TCP connection.
	  # The results of that transaction (successful or otherwise) are cached for the lifetime of that connection;
	  # at most one attempt to identify connection user is made.
	  #
	  # A configured dash ('-') value (or any equivalent regex) matches:
	  #    - A transaction that lacks access to a client-to-Squid connection (in the ACL evaluation context).
	  #      This happens if this ACL is used for transactions that failed to accept a TCP connection
	  #      and for transactions that were not triggered by an accepted TCP connection.
	  #      Squid logs an error message in these cases.
	  #    - A connection for which Squid has not obtained user identity yet.
	  #      This may happen if this ACL is used with a directive that does not support slow ACLs.
	  #      Such use cases do not trigger a transaction with an Authentication Server and do not
	  #      update the user-id cache.
	  #      Squid logs an error message in these cases.
	  #    - A connection for which Squid has failed to obtained user identity,
	  #      including cases where Authentication Server returned an invalid (e.g., empty) user-id.
	  #    - A connection for which Authentication Server returned a dash character as a user-id.
```

The above text may need to be fixed to match reality!'

Once we are done with `ident_regex`, we will polish `ident`. I hope that it can refer to  `ident_regex` for all cases other than the configured `REQUIRED` value.

Reply:
I posted an adjusted version at 9841f32 without the 1st and 2nd of your listed '-' cases. In both cases checklist->ident() is undefined and it does not attempt to match().
1. A transaction that lacks access to a client-to-Squid connection. This case results in level 1 message"ERROR: Cannot start ident lookup" and ACCESS_DUNNO.
2.  A connection for which Squid has not obtained user identity yet. A fast ACL makes hecklist->goAsync() fail, resulting in level 3 message "cannot start ident lookup" and ACCESS_DUNNO.

IMO the both cases indicate some general failure and we should not adjust the code to make them match '-'.